.github/workflows/ruby-build.yml

```diff
@@ -86,20 +86,23 @@ jobs:
             ruby/target/release/ruby-extractor.exe
           retention-days: 1
   compile-queries:
-    runs-on: ubuntu-latest
-    env:
-      CODEQL_THREADS: 4 # TODO: remove this once it's set by the CLI
+    runs-on: ubuntu-latest-xl
```

Comment:
I guess XL runners are a bit overkill , this job only compiles some Ruby queries, and most of them should be cached already.

Reply:
I still want to keep it.  
If the core libraries are changed the cache won't help much. 
And even when we get a full cache hit the XL worker is significantly faster.